### PR TITLE
add support for ignore radius from agent scripts

### PR DESCRIPTION
**Pull request by ZappoMan**
**Test plan:**
- Run this version of the sandbox
- In domain settings page -- add the following script to your scripts list: https://raw.githubusercontent.com/ZappoMan/hifi/8dc41ca8b9ed812a297fac86281c5f7ee595f85d/script-archive/acScripts/BetterClientSimulationBotFromRecording.js
- Go to the domain, and go to approximately 105,0,30 and look in the direction of ~-160 yaw... you should see a random bot running...
- Approach the bot... you should be able to get close to it without it disappearing.
- If you compare this to Master and the following script: https://raw.githubusercontent.com/highfidelity/hifi/master/script-archive/acScripts/BetterClientSimulationBotFromRecording.js you will notice that as you approach the bot, the space bubble engages and the bot disappears.

### CODE REVIEW

File: assignment-client/src/Agent.cpp [hunk 1]
```diff
@@ -351,6 +354,10 @@ void Agent::executeScript() {
     // give this AvatarData object to the script engine
     _scriptEngine->registerGlobalObject("Avatar", scriptedAvatar.data());
 
+    // give scripts access to the Users object
+    _scriptEngine->registerGlobalObject("Users", DependencyManager::get<UsersScriptingInterface>().data());
```
Comment (birarda) [inline assignment-client/src/Agent.cpp]: I'm concerned about broadly adding this to the Agent, would prefer to see us add `disableIgnoreRadius` to `ScriptableAvatar`

Comment (ZappoMan) [inline assignment-client/src/Agent.cpp]: checking with anti-harassment team about this.

Comment (birarda) [inline assignment-client/src/Agent.cpp]: Reviewed Agent permissions, doesn't have the ability to kick. This should be fine then.

Comment (ZappoMan) [inline assignment-client/src/Agent.cpp]: @birarda and I checked the code... it turns out as this is implemented no additional dangerous rights are provided... so this is is ok

**Comment by hifi-gustavo** [comment]
Build console output for commit: 8dc41ca8b9ed812a297fac86281c5f7ee595f85d available at: https://gist.github.com/25fdf7ed0a703f91277442fee59eb58f<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10283-8dc41ca.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10283-8dc41ca.dmg

**Pull request merged**